crux.el

```diff
@@ -456,23 +456,47 @@ See `file-attributes' for more info."
   (equal (crux-file-owner-uid filename)
          (user-uid)))
 
+(defun crux-already-root-p ()
+  (let ((remote-method (file-remote-p default-directory 'method))
+        (remote-user (file-remote-p default-directory 'user)))
+    (when (and remote-method
+               (or (member remote-method '("sudo" "su" "ksu" "doas"))
+                   (string= remote-user "root")))
+      (message "Already editing this file as root."))))
```

Comment:
I don't think that a predicate method should have side effects (like printing a message).

Reply:
Agreed.